### PR TITLE
Update Langium to 3.1.0

### PR DESCRIPTION
**Pull request by msujew**
Preparation for the 3.1.0 release.

### CODE REVIEW

File: packages/langium/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+## v3.1.0 (Jun. 2024)
+
+### General Improvements
+
+* Completion items now show the documentation of an element ([#1516](https://github.com/eclipse-langium/langium/pull/1516)).
+* Better support for text document LSP operations ([#1474](https://github.com/eclipse-langium/langium/pull/1474)).
+* Improve unknown file type handling in the language server ([#1455](https://github.com/eclipse-langium/langium/pull/1455), [#1492](https://github.com/eclipse-langium/langium/pull/1492)).
+* Emit an event after configuration section updates ([#1445](https://github.com/eclipse-langium/langium/pull/1445)).
+* Deleted files now have their diagnostics removed ([#1441](https://github.com/eclipse-langium/langium/pull/1441)).
+* A few improvements to the grammar language validation ([#1459](https://github.com/eclipse-langium/langium/pull/1459), [#1401](https://github.com/eclipse-langium/langium/pull/1401)).
```
Comment (spoenemann) [inline packages/langium/CHANGELOG.md:12]: I think grammar validation could be moved to the changelog of the vscode extension.

File: packages/langium/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+
+### Bug Fixes
+
+* Fixed cyclic formatter behavior ([#1550](https://github.com/eclipse-langium/langium/pull/1550)).
+* Fixed incorrect CST nodes after performing grammar actions ([#1547](https://github.com/eclipse-langium/langium/pull/1547)).
+* Fixed a few `Stream` idempotency issues ([#1545](https://github.com/eclipse-langium/langium/pull/1545)).
+* Fixed a bunch of grammar type system issues ([#1506](https://github.com/eclipse-langium/langium/pull/1506), [#1500](https://github.com/eclipse-langium/langium/pull/1500), [#1478](https://github.com/eclipse-langium/langium/pull/1478), [#1473](https://github.com/eclipse-langium/langium/pull/1473)).
```
Comment (spoenemann) [inline packages/langium/CHANGELOG.md:19]: Same with these grammar-related fixes?

Here's a list of PRs that affected the grammar: https://github.com/eclipse-langium/langium/pull/1401, https://github.com/eclipse-langium/langium/pull/1459, https://github.com/eclipse-langium/langium/pull/1473, https://github.com/eclipse-langium/langium/pull/1478, https://github.com/eclipse-langium/langium/pull/1500, https://github.com/eclipse-langium/langium/pull/1506, https://github.com/eclipse-langium/langium/pull/1541

**Review by spoenemann — COMMENTED** [review]
Thanks! I would move a few items to the langium-vscode package.

**Review by spoenemann — APPROVED** [review]

File: packages/langium/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
 
+## v3.1.0 (Jun. 2024)
+
+### General Improvements
+
+* Completion items now show the documentation of an element ([#1516](https://github.com/eclipse-langium/langium/pull/1516)).
+* Better support for text document LSP operations ([#1474](https://github.com/eclipse-langium/langium/pull/1474)).
+* Improve unknown file type handling in the language server ([#1455](https://github.com/eclipse-langium/langium/pull/1455), [#1492](https://github.com/eclipse-langium/langium/pull/1492)).
+* Emit an event after configuration section updates ([#1445](https://github.com/eclipse-langium/langium/pull/1445)).
+* Deleted files now have their diagnostics removed ([#1441](https://github.com/eclipse-langium/langium/pull/1441)).
```
Comment (cdietrich) [inline packages/langium/CHANGELOG.md:11]: maybe also add a hint on https://github.com/eclipse-langium/langium/pull/1407

Comment (cdietrich) [reply packages/langium/CHANGELOG.md:11]: and maybe also mention api changes / deprecations on service registry

Comment (msujew) [reply packages/langium/CHANGELOG.md:11]: Done 👍

Comment (cdietrich) [reply packages/langium/CHANGELOG.md:11]: Nice

**Pull request merged**